### ISSUE TITLE
Upgrade to RL-0.4.2,  Run for 1 day, but did not receive any rewards, there are transaction records。 Help ! Help ! Help !

### ISSUE THREAD

**Issue by fxtxkktv** [gensyn-ai/rl-swarm#267]
![Image](https://github.com/user-attachments/assets/81f2e8e9-a791-4fa8-8502-1055b2962df4)
![Image](https://github.com/user-attachments/assets/f4eb8e92-63ec-4964-bc9d-e254c1cf799d)

# Model arguments
model_revision: main
torch_dtype: float32
attn_implementation: eager
bf16: false
tf32: false

# Dataset arguments
dataset_id_or_path: 'openai/gsm8k'

# Training arguments
max_steps: 10 # Original 450
gradient_accumulation_steps: 2
gradient_checkpointing: false
gradient_checkpointing_kwargs:
use_reentrant: false
learning_rate: 3.0e-6 # 1.0e-6 1.0e-6 as in the deepseek math paper 5-e7 from https://hijkzzz.notion.site/unraveling-rlhf-and-its-variants-engineering-insights#147d9a33ecc9806090f3d5c749d31f05
lr_scheduler_type: cosine
warmup_ratio: 0.03

# GRPO arguments
use_vllm: false
num_generations: 2
per_device_train_batch_size: 2
beta: 0.01 # 0.04 as in the deepseek math paper 0.001 from https://hijkzzz.notion.site/unraveling-rlhf-and-its-variants-engineering-insights#147d9a33ecc9806090f3d5c749d31f05
max_prompt_length: 64
max_completion_length: 256

# CPU-specific optimizations
num_workers: 8  # = Number of physical cores (18 cores/CPU x 2 CPUs)
ddp_enabled: true
torch_distributed_backend: "gloo"  # Required when using CPU
torch_num_threads: 16  # Total threads (36 threads/CPU x 2 CPUs)
torch_intraop_threads: 8  # Each operation uses 1 CPU (36 threads)
torch_interop_threads: 2  # Keep this to avoid contention
pin_memory: false  # Not needed for CPU
non_blocking: false  # Only meaningful with GPU
prefetch_factor: 4  # Speed up data reading
torch_use_deterministic_algorithms: false  # Disable to speed up
torch_use_cpu_affinity: true  # Pin threads to physical cores

# IMPORTANT: NUMA-aware settings (Linux)
numactl_enabled: true           # Required to use both CPUs evenly
numactl_cpunodebind: "0,1"      # Bind process to both CPUs
torch_use_cpu_affinity: true    # Pin threads to physical cores
 
numactl_enabled: true         
numactl_cpunodebind: "0,1"     
torch_use_cpu_affinity: true   

# Logging arguments
logging_strategy: steps
logging_steps: 2
report_to:
- tensorboard
save_strategy: "steps"
save_steps: 25
seed: 42

# Script arguments
max_rounds: 10000

# Model-specific arguments
model_name_or_path: unsloth/Qwen2.5-0.5B-Instruct
output_dir: runs/gsm8k/multinode/Qwen2.5-0.5B-Instruct-Gensyn-Swarm

**Comment by fxtxkktv** [comment]
@tkocmathla @cleanunicorn @roman-kashitsyn @yihualou

**Comment by h-grieve** [comment]
Don't tag the team directly -- we'll review these and get back to you. Thanks

**Comment by cleanunicorn** [comment]
I see the node ran and reported rewards correctly.

Reopen if you have issues.

![Image](https://github.com/user-attachments/assets/4b65214c-ee7e-4bcf-98c6-ddcd2dd06e24)
